src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```diff
@@ -0,0 +1,638 @@
+    }
+
+    /**
+     * Disconnect from the active call or from an idle call (ZR is trying to connect)
+     *
+     * @return true if disconnected successfully, false if an error has occurred
+     */
+    private boolean callDisconnect() throws Exception {
+        if (isHost) {
+            return executeAndVerify(ZCOMMAND_CALL_DISCONNECT, "*r CallDisconnectResult");
+        } else {
+            return executeAndVerify(ZCOMMAND_CALL_LEAVE, "*r CallDisconnectResult");
+        }
+    }
+
+    /**
+     * Execute command using execute() method and simply check whether the command completed successfully
+     *
+     * @param command      to perform
+     * @param verifyString string that should be present in a response payload
+     * @return boolean value indicating whether the operation is successful
+     */
+    private boolean executeAndVerify(String command, String verifyString) throws Exception {
+        String response = execute(command, verifyString);
+        return !StringUtils.isNullOrEmpty(response);
+    }
+
+    /**
+     * Execute the command and check the response to contain "verifyString" in a few attempts.
+     * It's not added into the general "CommandSuccess/Failure" list because the basic communicator
+     * checks for endsWith() but there's no guarantee that the response will end with a certain string,
+     * also sometimes there are multiple commands performed automatically on entry with multiple "** end" breaks.
+     * It makes it impossible to verify whether the "** end" string was indicating the "real" end of a payload
+     * or not.
+     * So current scenario is:
+     *  - Command is sent
+     *  - Verifying if the response contains substring, specific for the current command
+     *      - If not - the communicator did not receive the response and we can't rely on any data in the response as an
+     *        indicator of an "end of output", re-issuing the command.
+     *      - If it contains the expected substring - OK, the response is unique and the communicator has reached
+     *        an end of output.
+     *
+     * @param command      to perform
+     * @param verifyString string that should be present in a response payload
+     * @return empty string or a command response, if successful
+     */
+    private String execute(String command, String verifyString) throws Exception {
+        String response;
+        controlOperationsLock.lock();
+        try {
+            refreshSshConnection();
+            response = send(command);
+            int retryAttempts = 0;
+            while (!response.contains(verifyString)) {
+                retryAttempts++;
+                response = send(command);
+                if (retryAttempts >= 10) {
+                    throw new IllegalStateException(String.format("Failed to verify response for command %s. Expected output: %s", command, verifyString));

```

Comment:
Wrong exception (nothing illegal with communicator state at this moment). Please see similar comment above.